src/string-class.js

```diff
@@ -0,0 +1,191 @@
+const stringManipulations = {
+  /**
+   * hasVowels method test if a string contains vowels
+   * @return {boolean} true if there are vowels and false otherwise
+   */
+  hasVowels() {
+    return /[aeiou]/i.test(this);
+  },
+
+  /**
+   * toupper method converts all characters to their upper case
+   * @return {string}, a string with upper case characters
+   */
+  toUpper() {
+    return this.replace(/[a-z]/g, found =>
+      // makes use of ascii code
+      String.fromCharCode(found.charCodeAt(0) - 32)
+    );
+  },
+
+  /**
+   * toLower method converts all characters to their lower case
+   * @return {string}, a string with lower case characters
+   */
+  toLower() {
+    return this.replace(/[A-Z]/g, found =>
+      // makes use of ascii code
+      String.fromCharCode(found.charCodeAt(0) + 32)
+    );
+  },
+
+  /**
+   * ucFirst method converts the first character to upper case
+   * @return {string}, a string with it's initial character in upper case'
+   */
+  ucFirst() {
+    return this.replace(this.charAt(0), this.charAt(0).toUpper());
+  },
+
+  /**
+   * isQuestion method checks if a string is a question by
+   * checking if the last character is a question mark ?
+   * @return {boolean}; true if the string is a question and false otherwise
+   */
+  isQuestion() {
+    return /\?$/.test(this.trimSpace());
+  },
+
+  /**
+   * trimSpace method removes the trailing and preceeding empty
+   * spaces in a string
+   * @return {string}; trimmed string
+   */
+  trimSpace() {
+    return this.replace(/^\s+|\s+$/g, '');
+  },
+
+  /**
+   * words method gets all the words in a string
+   * @return {array}; an array of words in the string
+   */
+  words() {
+    const newValue = this.trimSpace().removeSpecialChars();
+    return newValue === '' ? [] : newValue.split(/\s+/);
+  },
+
+  /**
+   * removeSpecialChars method is used to replace non-alphanumeric characters
+   * including _ that are alone and replaces it with empty space
+   * @return {string}:
+   */
+  removeSpecialChars() {
+    return this.replace(/[^\w\s]|_/g, '');
+  },
+
+
+  /**
+   * wordCount method counts the number of words in a string
+   * @return {int}: number of words
+   */
+  wordCount() {
+    return this.words().length;
+  },
+
+  /**
+   * toCurrency method formats a number by putting commas in the right place
+   * @return {string}: string representing the currency format
+   */
+  toCurrency() {
+    const parts = this.split(/[.]/g),
+      length = parts[0].length;
+    let firstCommaPosition = length % 3;
+    firstCommaPosition = (firstCommaPosition === 0) ? 3 : firstCommaPosition;
+    /* eslint-disable prefer-template */
+    const regExpression =
+    new RegExp('^[0-9]{' + firstCommaPosition + '}|[0-9]{3}(?!$)', 'g');
```

Comment:
I believe we talked about using template literals for this operation. Did you encounter any challenges doing this?